Sources/Amplitude/Plugins/iOS/UIKitScreenViews.swift

```diff
@@ -0,0 +1,120 @@
+import Foundation
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+
+class UIKitScreenViews {
+    internal static let lock = NSLock()
+    internal static var amplitudes: [Amplitude] = []
+    private static var viewDidAppearSwizzled = false
+
+    static func register(_ amplitude: Amplitude) {
+        lock.lock()
+        defer { lock.unlock() }
+
+        amplitudes.append(amplitude)
+        if viewDidAppearSwizzled {
+            return
+        }
+
+        viewDidAppearSwizzled = true
+        swizzleViewDidAppear()
+    }
+
+    private static func swizzleViewDidAppear() {
+        let controllerClass = UIViewController.self
+        let originalSelector = #selector(UIViewController.viewDidAppear(_:))
```

Comment:
Are these selectors the same for all platforms (iOS, Mac, Watch)?

Reply:
What happens if this is not available e.g. WatchOS
https://developer.apple.com/documentation/uikit/uiviewcontroller/1621423-viewdidappear?language=objc

Reply:
Good catch, I think Mac is different, the class is [`NSViewController`](https://developer.apple.com/documentation/appkit/nsviewcontroller). iOS, tvOS, macCatalyst are technically supported, think the file already filters it with the macro. It only uses in `IOSLifecycleMonitor` in this PR right now. I suggest we only support iOS (and compatible series) for now, and expand to others if needed in the future. What do you think?